apps/er-coap/er-coap.h

```diff
@@ -156,18 +156,18 @@ typedef struct {
   }
 #define COAP_SERIALIZE_STRING_OPTION(number, field, splitter, text) \
   if(IS_OPTION(coap_pkt, number)) { \
-    PRINTF(text " [%.*s]\n", coap_pkt->field##_len, coap_pkt->field); \
+    PRINTF(text " [%.*s]\n", (int)coap_pkt->field##_len, coap_pkt->field); \
     option += coap_serialize_array_option(number, current_number, option, (uint8_t *)coap_pkt->field, coap_pkt->field##_len, splitter); \
     current_number = number; \
   }
 #define COAP_SERIALIZE_BLOCK_OPTION(number, field, text) \
   if(IS_OPTION(coap_pkt, number)) \
   { \
-    PRINTF(text " [%lu%s (%u B/blk)]\n", coap_pkt->field##_num, coap_pkt->field##_more ? "+" : "", coap_pkt->field##_size); \
+    PRINTF(text " [%lu%s (%u B/blk)]\n", (unsigned long)coap_pkt->field##_num, coap_pkt->field##_more ? "+" : "", coap_pkt->field##_size); \
     uint32_t block = coap_pkt->field##_num << 4; \
     if(coap_pkt->field##_more) { block |= 0x8; } \
     block |= 0xF & coap_log_2(coap_pkt->field##_size / 16); \
-    PRINTF(text " encoded: 0x%lX\n", block); \
+    PRINTF(text " encoded: 0x%lX\n", (unsigned long)block);		\
```

Comment:
I think %x always prints the value as unsigned. The only difference between %x and %X is that one prints lowercase and the other uppercase.